### PR TITLE
One readme per day

### PR DESCRIPTION
**Pull request by gmarcos87**
## Our packages lack documentation!!!

Only 5 of them have a readme and are probably outdated.
My proposal is to add one readme per package, one per day. I think it's a good practice, that it won't consume a lot of time and that it will allow a better understanding of our ecosystem.

ezgif-5-13d655870a4d

Those who want to participate are invited. This is the package list:

- [x] auto-usb-wwan
- [x] bandwidth-test
- [x] batman-adv-auto-gw-mode
- [ ] bmx6-auto-gw-mode
- [ ] bmx7-auto-gw-bw-mode
- [ ] bmx7-auto-gw-mode
- [ ] bmx7-mdns
- [x] check-date-http
- [ ] first-boot-wizard
- [ ] hotplug-initd-services
- [ ] lime-app
- [ ] lime-ap-watchping
- [ ] lime-babel
- [ ] lime-basic
- [ ] lime-basic-uing
- [ ] lime-debug
- [ ] lime-docs
- [ ] lime-hwd-ground-routing
- [ ] lime-hwd-openwrt-wan
- [ ] lime-hwd-usbradio
- [ ] lime-map-agent
- [ ] lime-proto-anygw
- [ ] lime-proto-babeld
- [ ] lime-proto-batadv
- [ ] lime-proto-bgp
- [ ] lime-proto-bmx6
- [ ] lime-proto-bmx7
- [ ] lime-proto-olsr
- [ ] lime-proto-olsr2
- [ ] lime-proto-olsr6
- [ ] lime-proto-wan
- [ ] lime-smart-wifi
- [ ] lime-system
- [ ] lime-webui
- [x] lua-argparse
- [ ] luci-app-batman-adv
- [ ] luci-app-lime-location
- [ ] luci-app-openairview
- [ ] luci-lib-jquery-flot
- [ ] luci-mod-status
- [x] pirania
- [ ] safe-reboot
- [x] safe-upgrade
- [ ] shared-state
- [ ] shared-state-babeld_hosts
- [ ] shared-state-bat_hosts
- [ ] shared-state-dnsmasq_hosts
- [ ] shared-state-dnsmasq_leases
- [ ] shared-state-nodes_and_links
- [ ] shared-state-persist
- [ ] shared-state-pirania
- [ ] soporteremoto
- [ ] sprunge
- [x] tinc
- [ ] ubus-lime-batman-adv
- [ ] ubus-lime-bmx6
- [ ] ubus-lime-fbw
- [ ] ubus-lime-groundrouting
- [ ] ubus-lime-location
- [ ] ubus-lime-metrics
- [x] ubus-lime-openairview
- [x] ubus-lime-utils
- [ ] watchping

Ignored or deprecated
- ~~altermap-agent~~
- ~~vale~~
- [x] ~~cotonete~~
- ~~dnsmasq-distributed-hosts~~
- ~~dnsmasq-lease-share~~
- ~~fft-eval~~
- ~~smoni~~

### CODE REVIEW

**Comment by ilario** [comment]
I never noticed that any of the packages had a readme file in their directory.
I mean: I think people does not look for documentation in there, isn't there a better place?
What about improving the `description` field in the package's Makefile and creating documentation on the website?

**Comment by gmarcos87** [comment]
In the description field we cannot use markdown, and its visibility is more conditioned to the use of "make menuconfig".
We can extract the readmes from lime-packages to generate a documentation section on the web, that would be interesting.

**Comment by spiccinini** [comment]
I also think that is better to have a Readme.md for each package and have a short description of the package at the Makefile description because it is easier to add more information and is easy to read from a Readme. Wwe can even add _see the package/Readme for more documentation_ at the Makefile description.
Great initiative :)

**Comment by ilario** [comment]
>  We can extract the readmes from lime-packages to generate a documentation section on the web, that would be interesting.

...sounds good, how could we achieve this?

> We can even add for more details read the package/Readme.md at the Makefile description.

This helps :)

File: packages/bandwidth-test/Readme.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+## bandwidth-test
```
Comment (ilario) [inline packages/bandwidth-test/Readme.md]: We could point out that running `bandwidth-test -h` (actually also `bandwidth-test hakuna matata` gives the same output, but `-h` looks more professional) outputs an usage help.
Or we could just copy and paste here the help text which can be taken from [here](https://github.com/libremesh/lime-packages/blob/master/packages/bandwidth-test/files/bin/bandwidth-test.lua#L47), what do you prefer?

Comment (gmarcos87) [reply packages/bandwidth-test/Readme.md]: In my case "bandwidth-test -h" triggers the measurements, it doesn't show the help.

`root@ql-anaymarcos:~# bandwidth-test -h
Attempting connection to http://speedtest-lon1.digitalocean.com/10mb.test
`

Comment (ilario) [reply packages/bandwidth-test/Readme.md]: LOL sorry, I'm going to fix that XD

**Comment by G10h4ck** [comment]
Some of the sections of this document I wrote a bunch of years ago http://nicolasacco.diveni.re/~gio/internship-report/main.html  may be a little outdated but most of it is not and reading it would clarify the relation between most of the lime-* packages

**Comment by gmarcos87** [comment]
Okay, this is gonna be impossible.
I'd better pull now and resume this task in the near future. It was worth a try

**Pull request merged**